File: 485-max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int n=nums.length;
        int flag=0;
        int arr[]=new int[n];
        int j=0;
        for(int i=0;i<n;i++){
            if(nums[i]==1){
                flag++;
                if(i==n-1){
                    arr[j++]=flag;
                }
            }
            else {
                arr[j++]=flag;
                flag=0;
            }
        }
        int lrg=0;
        for(int i=0;i<arr.length;i++){
            if(lrg<arr[i]){
                lrg=arr[i];
            }
        }
        return lrg;
    }
}

File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n=nums.size();
        int mx=0, curr=0;
        for(int i=0;i<n;i++){
            if(nums[i]!=1){
                curr=0;
            }
            else curr++;
            if(curr>mx) mx=curr;
        }
        return mx;
    }
};